.github/workflows/nightly.yml

```diff
@@ -0,0 +1,229 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# ---------------------------------------------------------
+name: nightly
+
+on:
+  #          ┌───────────── minute (0 - 59)
+  #          │ ┌───────────── hour (0 - 23)
+  #          │ │ ┌───────────── day of the month (1 - 31)
+  #          │ │ │ ┌───────────── month (1 - 12 or JAN-DEC)
+  #          │ │ │ │ ┌───────────── day of the week (0 - 6 or SUN-SAT)                                  
+  #          │ │ │ │ │
+  #          │ │ │ │ │
+  schedule: 
+    - cron: '0 0 * * *'  # basically running everyday at 12AM
+    # cron works with default branch (main) only: # https://github.community/t/on-schedule-per-branch/17525/2
+  
+  push:
+    branches:
+      # Because we can't schedule run for non-main branches,
+      # to ensure we are running the build on the staging branch, we can add push policy for it
+      - staging
+      # Development triggers, can be removed afterwards
+      - laserprec/nightly-ci
+
+# This file defines following CI workflow:
+#
+#  ┌──────────┐   ┌─────────┐   ┌────────┐
+#  │  static  ├─┬─►  build* ├─┬─►  test  │
+#  │ analysis │ │ │  (cpu)  │ │ │ report │
+#  └──────────┘ │ └─────────┘ │ └────────┘         
+#               │ ┌─────────┐ │
+#               ├─►  build* ├─┤
+#               │ │ (spark) │ │
+#               │ └─────────┘ │
+#               │ ┌─────────┐ │
+#               └─►  build* ├─┘
+#                 │  (gpu)  │
+#                 └─────────┘
+#                    ....
+# *each runs in PARALLEL the different combinations
+#  of python version, OS, test subsets, etc
+#
+#  ASCII chart created via https://asciiflow.com/
+
+jobs:
+###############################################
+############### STATIC-ANALYSIS ###############
+###############################################
+  static-analysis:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Use Python 3.6
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.6
+    
+    - name: Install dependencies (tox)
+      run: |
+        python -m pip install --upgrade pip setuptools wheel
+        pip install tox
+
+    - name: Run flake8
+      # TODO: re-enable this flake8 block (turned off to get a draft of the pipeline infrastructure)
+      continue-on-error: true
+      run: |
+       tox -e flake8
+
+###############################################
+################## CPU-BUILD ##################
+###############################################
+  build-cpu:
+    runs-on: ${{ matrix.os }}
+    needs: static-analysis
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+        python: [3.6]
```

Comment:
let's say we wanted to add python 3.7 in parallel, how will that look like?

Reply:
we can just expand the python array to:

```
matrix:
   os: [ubuntu-latest]
   python: [3,6,, 3.7]
```

Reply:
can we try to add both 3.6 and 3.7? My guess is that the current code would work with both. I'm assuming that each pipeline will run in parallel, right?

Reply:
There is an issue I have observed with python 3.7. When pip installing with the gpu option (on a DSVM) and importing numba I get
numpy.core.multiarray failed to import

Reply:
Looking at SOF, it seems that this is solved by numpy 1.8 https://stackoverflow.com/a/21946788 , but we have a higher numpy. If this is the only issue we have in python 3.7, it might be worthwhile explore if just a change in the version of numpy or numba fixes the problem